File: app/src/main/java/net/a8pade8/passwordsaver/AddRecordActivity.java
package net.a8pade8.passwordsaver;

import android.content.DialogInterface;
import android.support.v7.app.AlertDialog;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.text.InputType;
import android.text.method.HideReturnsTransformationMethod;
import android.text.method.PasswordTransformationMethod;
import android.view.View;
import android.widget.EditText;
import android.widget.ToggleButton;

import net.a8pade8.passwordsaver.a8pade8Lib1.Messages;
import net.a8pade8.passwordsaver.data.DbserviceKt;
import net.a8pade8.passwordsaver.data.EmptyDataException;

public class AddRecordActivity extends AppCompatActivity {

    private EditText editTextResource, editTextLogin, editTextPassword, editTextPasswordReplay;
    private ToggleButton buttonSite, buttonFade, buttonEmail;
    private int inputTypeLogin;
    private int inputTypeResource;

    private String resource;
    private String login;
    private String password;
    private String passwordRetry;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_add_record);
        editTextLogin = findViewById(R.id.editTextLogin);
        editTextPassword = findViewById(R.id.editTextPassword);
        editTextPasswordReplay = findViewById(R.id.editTextPasswordReplay);
        editTextResource = findViewById(R.id.editTextResourse);

        buttonEmail = findViewById(R.id.toggleButtonEmail);
        buttonFade = findViewById(R.id.toggleButtonFade);
        buttonSite = findViewById(R.id.toggleButtonSite);

        inputTypeLogin = editTextLogin.getInputType();
        inputTypeResource = editTextResource.getInputType();
    }

    public void onSwitchSite(View view) {
        if (buttonSite.isChecked()) {
            editTextResource.setInputType(InputType.TYPE_CLASS_TEXT);
        } else {
            editTextResource.setInputType(inputTypeResource);
        }
    }

    public void onSwitchEmail(View view) {
        if (buttonEmail.isChecked()) {
            editTextLogin.setInputType(InputType.TYPE_CLASS_TEXT);
        } else {
            editTextLogin.setInputType(inputTypeLogin);
        }
    }

    public void onSwitchFade(View view) {
        if (!buttonFade.isChecked()) {
            editTextPassword.setTransformationMethod(PasswordTransformationMethod.getInstance());
            editTextPasswordReplay.setTransformationMethod(PasswordTransformationMethod.getInstance());
        } else {
            editTextPassword.setTransformationMethod(HideReturnsTransformationMethod.getInstance());
            editTextPasswordReplay.setTransformationMethod(HideReturnsTransformationMethod.getInstance());
        }
    }

    public void onReady(View view) {

        password = editTextPassword.getText().toString();
        passwordRetry = editTextPasswordReplay.getText().toString();
        login = editTextLogin.getText().toString();
        resource = editTextResource.getText().toString();

        if (isValuesChecked() && isPasswordsEquals()) {

            if (!isResourceExist()) {
                addRecord();
            } else {
                messageDoubleResource();
            }
        }
    }

    private boolean isPasswordsEquals() {
        if (!password.equals(passwordRetry)) {
            Messages.MiddleToastShort(this, "Пароли не совпадают.");
            return false;
        } else {
            return true;
        }
    }

    private boolean isResourceExist() {
        return DbserviceKt.isContainResourceInPasswords(resource);
    }


    private void messageDoubleResource() {

        AlertDialog.Builder ad = new AlertDialog.Builder(this);
        ad.setTitle("Внимание!");
        ad.setMessage("Указанный ресурс уже существует. Все равно добавить?");
        ad.setCancelable(false);
        ad.setPositiveButton("Добавить", (dialogInterface, i) -> {
            addRecord();
            dialogInterface.cancel();
        });
        ad.setNegativeButton("Отмена", (dialogInterface, i) -> dialogInterface.cancel());
        AlertDialog alert = ad.create();
        alert.show();
    }

    private boolean isValuesChecked() {
        return isResourceEmpty()
                && isLoginEmpty()
                && isPasswordEmpty();
    }

    private boolean isResourceEmpty() {
        if (editTextResource.getText().toString().equals("")) {
            Messages.MiddleToastShort(this, "Поле ввода РЕСУРС не заполнено.");
            return false;
        } else {
            return true;
        }
    }

    private boolean isLoginEmpty() {
        if (editTextLogin.getText().toString().equals("")) {
            Messages.MiddleToastShort(this, "Поле ввода ЛОГИН не заполнено.");
            return false;
        } else {
            return true;
        }
    }

    private boolean isPasswordEmpty() {
        if (editTextPassword.getText().toString().equals("")) {
            Messages.MiddleToastShort(this, "Поле ввода ПАРОЛЬ не заполнено.");
            return false;
        }
        return true;
    }

    private void addRecord() {
        try {
            DbserviceKt.addRecordToPasswords(resource, login, password);
        } catch (EmptyDataException e) {
            Messages.MiddleToastLong(this, "Ошибка, данные не указаны");
            e.printStackTrace();
        }
        Messages.MiddleToastLong(this, "Запись успешно добавлена.");
        this.finish();
    }
}

File: app/src/main/java/net/a8pade8/passwordsaver/activities/AddRecordActivity.kt
package net.a8pade8.passwordsaver.activities

import android.app.AlertDialog
import android.content.DialogInterface
import android.os.Bundle
import android.text.InputType.TYPE_CLASS_TEXT
import android.text.InputType.TYPE_TEXT_VARIATION_URI
import android.text.method.HideReturnsTransformationMethod
import android.text.method.PasswordTransformationMethod
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import kotlinx.android.synthetic.main.activity_add_record.*
import net.a8pade8.passwordsaver.R
import net.a8pade8.passwordsaver.R.string.*
import net.a8pade8.passwordsaver.data.*
import net.a8pade8.passwordsaver.databinding.ActivityAddRecordBinding
import net.a8pade8.passwordsaver.uiutil.showShortSnack
import net.a8pade8.passwordsaver.util.generateAlthaNumericString

class AddRecordActivity : AppCompatActivity() {

    private lateinit var binding: ActivityAddRecordBinding
    private val TYPE_AUTO_COMPLETE_EMAIL = 65569
    private var record = Record(0, "", "", "", "")

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_add_record)
        binding.record = record
        initToolbar()
    }

    private fun initToolbar() {
        setSupportActionBar(binding.mainToolbar)
    }

    @Suppress("UNUSED_PARAMETER")
    fun onSwitchSite(view: View) {
        if (toggleButtonSite.isChecked) {
            editTextLogin.inputType = TYPE_CLASS_TEXT
        } else {
            editTextLogin.inputType = TYPE_TEXT_VARIATION_URI
        }
    }

    @Suppress("UNUSED_PARAMETER")
    fun onSwitchEmail(view: View) {
        if (toggleButtonEmail.isChecked) {
            editTextLogin.inputType = TYPE_CLASS_TEXT
        } else {
            editTextLogin.inputType = TYPE_AUTO_COMPLETE_EMAIL
        }
    }

    @Suppress("UNUSED_PARAMETER")
    fun onSwitchFade(view: View) {
        if (!toggleButtonFade.isChecked) {
            editTextPassword.transformationMethod = PasswordTransformationMethod.getInstance()
            editTextPasswordReplay.transformationMethod = PasswordTransformationMethod.getInstance()
        } else {
            editTextPassword.transformationMethod = HideReturnsTransformationMethod.getInstance()
            editTextPasswordReplay.transformationMethod = HideReturnsTransformationMethod.getInstance()
        }
    }

    @Suppress("UNUSED_PARAMETER")
    fun onReady(view: View) {
        if (isValuesChecked() && isPasswordsEquals()) {
            if (!isResourceExist()) {
                addRecord()
            } else {
                messageDoubleResource()
            }
        }
    }

    @Suppress("UNUSED_PARAMETER")
    fun generatePassword(view: View) {
        generateAlthaNumericString(6).let {
            record.password = it
            binding.passwordRetry = it
            binding.invalidateAll()
        }
    }

    private fun isPasswordsEquals(): Boolean {
        return if (record.password != binding.passwordRetry) {
            showShortSnack(getString(passwordsNotEquals))
            false
        } else {
            true
        }
    }

    private fun isResourceExist(): Boolean {
        return isContainResourceInPasswords(record.resourceName)
    }

    private fun messageDoubleResource() {
        AlertDialog.Builder(this)
                .setTitle(getString(warning))
                .setMessage(getString(repeatResource))
                .setCancelable(false)
                .setPositiveButton(getString(add)) { dialogInterface: DialogInterface?, i: Int -> addRecord() }
                .setNegativeButton(getString(cancel)) { dialogInterface: DialogInterface, i: Int -> dialogInterface.cancel() }
                .create()
                .show()
    }

    private fun isValuesChecked(): Boolean {
        return (isResourceNotEmpty()
                && isLoginNotEmpty()
                && isPasswordNotEmpty())
    }

    private fun isResourceNotEmpty(): Boolean {
        return if (record.resourceName.isEmpty()) {
            showShortSnack(getString(emptyResourceName))
            false
        } else {
            true
        }
    }

    private fun isLoginNotEmpty(): Boolean {
        return if (record.login.isEmpty()) {
            showShortSnack(getString(emptyLogin))
            false
        } else {
            true
        }
    }

    private fun isPasswordNotEmpty(): Boolean {
        if (record.password.isEmpty()) {
            showShortSnack(getString(emptyPassword))
            return false
        }
        return true
    }

    private fun addRecord() {
        try {
            addRecordToPasswords(record.resourceName, record.login, record.password, record.comment)
            showShortSnack(getString(recordAddedSucsessfully))
            finish()
        } catch (e: EmptyDataException) {
            showShortSnack(getString(dataError))
            e.printStackTrace()
        } catch (e: ResourceLoginRepeatException) {
            showShortSnack(getString(repeatLoginResource))
            e.printStackTrace()
        }
    }
}